cli/chaosli/cmd/context.go

```diff
@@ -0,0 +1,432 @@
+	// mention to the user that the labels they are using won't target anything
+
+	if size <= 0 {
+		errorString := fmt.Sprintf("\nThe label selectors chosen (%s) result in 0 targets, meaning this disruption would do nothing given the namespace/cluster/label combination.", labels)
+
+		if level == types.DisruptionLevelPod {
+			disruption.Spec.Level = types.DisruptionLevelNode
+		} else {
+			disruption.Spec.Level = types.DisruptionLevelPod
+		}
+
+		size = getTargetSize(disruption)
+
+		if size > 0 {
+			errorString = fmt.Sprintf("\nWe noticed that your target size is 0 for level %s given your label selectors. We checked to see if the %s level would give you results and we found %d %ss. Is this the level you wanted to use?", level, disruption.Spec.Level, size, disruption.Spec.Level)
+		}
+
+		return nil, nil, fmt.Errorf(errorString)
+	}
+
+	if level == types.DisruptionLevelPod {
+		allPods = showPods(pods)
+	} else {
+		allNodes = showNodes(nodes)
+	}
+
+	PrintSeparator()
+
+	return allPods, allNodes, nil
+}
+
+func showPods(pods v1.PodList) []v1.Pod {
+	targetsShow := []string{}
+
+	targetsAll := make([]v1.Pod, len(pods.Items))
+
+	for i, pod := range pods.Items {
+		if len(targetsShow) < maxtargetshow {
+			targetsShow = append(targetsShow, pod.Name)
+		}
+
+		targetsAll[i] = pod
+	}
+
+	fmt.Printf("\n🎯 There are %d pods that will be targeted\n", len(pods.Items))
+
+	if len(pods.Items) > maxtargetshow {
+		fmt.Println("📝 Here are a small set of those targets:")
+	}
+
+	for _, target := range targetsShow {
+		fmt.Println(target)
+	}
+
+	if len(pods.Items) > maxtargetshow {
+		fmt.Println("...")
+	}
+
+	return targetsAll
+}
+
+func showNodes(nodes v1.NodeList) []v1.Node {
+	targetsShow := []string{}
+	targetsAll := []v1.Node{}
+
+	for _, node := range nodes.Items {
+		if len(targetsShow) < maxtargetshow {
+			targetsShow = append(targetsShow, node.Name)
+		}
+
+		targetsAll = append(targetsAll, node)
+	}
+
+	fmt.Printf("\n🎯 There are %d nodes that will be targeted\n", len(nodes.Items))
+
+	if len(nodes.Items) > maxtargetshow {
+		fmt.Println("📝 Here are a small set of those targets:")
+	}
+
+	for _, target := range targetsShow {
+		fmt.Println(target)
+	}
+
+	if len(nodes.Items) > maxtargetshow {
+		fmt.Println("...")
+	}
+
+	return targetsAll
+}
+
+func getTargetSize(disruption v1beta1.Disruption) int {
+	level := disruption.Spec.Level
+	size := 0
+
+	if level == types.DisruptionLevelPod {
+		size = len(getPods(disruption).Items)
+	} else {
+		size = len(getNodes(disruption).Items)
+	}
+
+	return size
+}
+
+func getPods(disruption v1beta1.Disruption) v1.PodList {
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	options := metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(disruption.Spec.Selector).String(),
+	}
```

Comment:
Those lines are common to `getPods` and `getNodes` and should probably be part of the [command initialization](https://pkg.go.dev/github.com/spf13/cobra#OnInitialize).

We should also avoid `panic()` especially on errors which seems could be triggered easily (if the k8s configuration is not present, not correct, etc.) and should display a nice to read message instead so users are not lost in a stacktrace.

Reply:
Im not sure if OnInitialize is what we want to use because its in the root.go file and Im not sure how to nicely bring kubeconfig into context.go. But I added a setKubeconfig function in the context.go file that aggregates all configuration setup for kubeconfig. Let me know what you think.

Reply:
LGTM